app/api/groups/update_group_test.go

```diff
@@ -0,0 +1,137 @@
+	assertUpdateGroupFailsOnDBErrorInTransaction(t, func(mock sqlmock.Sqlmock) {
+		mock.ExpectBegin()
+		mock.ExpectQuery(`SELECT .+ WHERE \(users\.ID = \?\)`).WithArgs(2).WillReturnError(errors.New("error"))
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT groups.bFreeAccess FROM `groups` JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID WHERE (groups_ancestors.idGroupAncestor=?) AND (groups.ID = ?) LIMIT 1 FOR UPDATE")).
+			WithArgs(0, 1).WillReturnError(errors.New("some error"))
+		mock.ExpectRollback()
+	})
+}
+
+func TestService_updateGroup_ErrorOnRefusingSentGroupRequests(t *testing.T) {
+	assertUpdateGroupFailsOnDBErrorInTransaction(t, func(mock sqlmock.Sqlmock) {
+		mock.ExpectBegin()
+		mock.ExpectQuery(`SELECT .+ WHERE \(users\.ID = \?\)`).WithArgs(2).
+			WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(2))
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT groups.bFreeAccess FROM `groups` JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID WHERE (groups_ancestors.idGroupAncestor=?) AND (groups.ID = ?) LIMIT 1 FOR UPDATE")).
+			WithArgs(0, 1).WillReturnRows(sqlmock.NewRows([]string{"bFreeAccess"}).AddRow(true))
+		mock.ExpectExec("UPDATE `groups_groups` .+").WithArgs("requestRefused", 1).
+			WillReturnError(errors.New("some error"))
+		mock.ExpectRollback()
+	})
+}
+
+func TestService_updateGroup_ErrorOnUpdatingGroup(t *testing.T) {
+	assertUpdateGroupFailsOnDBErrorInTransaction(t, func(mock sqlmock.Sqlmock) {
+		mock.ExpectBegin()
+		mock.ExpectQuery(`SELECT .+ WHERE \(users\.ID = \?\)`).WithArgs(2).
+			WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(2))
+		mock.ExpectQuery(regexp.QuoteMeta("SELECT groups.bFreeAccess FROM `groups` JOIN groups_ancestors ON groups_ancestors.idGroupChild = groups.ID WHERE (groups_ancestors.idGroupAncestor=?) AND (groups.ID = ?) LIMIT 1 FOR UPDATE")).
+			WithArgs(0, 1).WillReturnRows(sqlmock.NewRows([]string{"bFreeAccess"}).AddRow(false))
+		mock.ExpectExec("UPDATE `groups` .+").
+			WillReturnError(errors.New("some error"))
+		mock.ExpectRollback()
+	})
+}
+
+func assertUpdateGroupFailsOnDBErrorInTransaction(t *testing.T, setMockExpectationsFunc func(sqlmock.Sqlmock)) {
+	db, mock := database.NewDBMock()
+	defer func() { _ = db.Close() }()
+
+	setMockExpectationsFunc(mock)
+
+	base := service.Base{Store: database.NewDataStore(db), Config: nil}
+	srv := &Service{Base: base}
+	router := chi.NewRouter()
+	router.Post("/groups/{group_id}", service.AppHandler(srv.updateGroup).ServeHTTP)
+
+	monkey.Patch(auth.UserIDFromContext, func(context context.Context) int64 {
+		return 2
+	})
+	defer monkey.UnpatchAll()
+
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	request, _ := http.NewRequest("POST", ts.URL+"/groups/1", strings.NewReader(`{"free_access":false}`))
+	response, err := http.DefaultClient.Do(request)
+
+	assert.NoError(t, err)
+	assert.Equal(t, 500, response.StatusCode)
+	assert.NoError(t, mock.ExpectationsWereMet())
+}
```

Comment:
😯This kind of test file is a good advocate for accepting `panic` for db error (unhandled). Writing a unit test for each DB request to simulate the DB failure on this particular request (without prior reason) is very time-consuming (and probably boring). 
If the db error would panic and be handled in the service.ServeHTTP, would you be able to avoid such unit tests for services?

Reply:
Yeah, it takes some time to write such tests. We would be able to avoid them if we use something like 
```
groupStore.OwnedBy(user).
			Select("groups.bFreeAccess").
			Set("gorm:query_option", "FOR UPDATE").
			Where("groups.ID = ?", groupID).Limit(1).Scan(&currentGroupData).MustBeNoError()
```
(which panics on errors) and service.ServeHTTP handles that. Do you want me to implement that in this PR?

Reply:
Oh great idea this function. Not in this PR for clarity please... (and it may raise some more debates, so better in another PR once this one is merged in master). An alternative naming: "MustNotFail".
🤔 Still... it would be a panic cross package :-/ Or maybe something like a `service.PanicOnFail(Failable)` where Failable is an interface implementing `Error() error` It would allow you to write:

    db := groupStore.OwnedBy(user).....Scan(&currentGroupData) 
    service.PanicOnError(db)